### PR TITLE
Support NVMEM layout for BSEC

### PR DESCRIPTION
**Pull request by GseoC**
Support the NVMEM layout that is used to describe OTP mapping in the device-tree.

This allows drivers to find an OTP location (BSEC word number and bit size) from the OTP string identifier, name of the cell in device tree. No more hardcoded OTP index :)

### CODE REVIEW

File: core/include/drivers/stm32_bsec.h [hunk 1]
```diff
@@ -147,4 +147,15 @@ TEE_Result stm32_bsec_otp_lock(uint32_t service);
  */
 bool stm32_bsec_nsec_can_access_otp(uint32_t otp_id);
 
+/*
+ * Find and get OTP location from its name.
+ * @name: sub-node name to look up.
+ * @otp_id: pointer to read OTP number or NULL.
+ * @otp_bit_len: pointer to read OTP length in bits or NULL.
```
Comment (etienne-lms) [inline core/include/drivers/stm32_bsec.h:154]: s/read/output/ :
```
 * @otp_id: pointer to output OTP number or NULL.
 * @otp_bit_len: pointer to output OTP length in bits or NULL.

**Review by etienne-lms — COMMENTED** [review]
`Reviewed-by: Etienne Carriere <etienne.carriere@linaro.org>` for commits.
"drivers: stm32_bsec: support NVMEM cell description in device-tree"
and "plat-stm32mp1: use DT NVMEM layout API", with comments addressed or not.

`Acked-by: Etienne Carriere <etienne.carriere@linaro.org>` for commit:
"dts: stm32: add nvmem layout".

File: core/drivers/stm32_bsec.c [hunk 2]
```diff
@@ -631,6 +670,72 @@ static void bsec_dt_otp_nsec_access(void *fdt, int bsec_node)
...
+		string = fdt_get_name(fdt, node, &len);
+		if (!string || !len)
+			continue;
+
+		reg_offset = _fdt_reg_base_address(fdt, node);
+		reg_length = _fdt_reg_size(fdt, node);
+
+		if (reg_offset == DT_INFO_INVALID_REG ||
+		    reg_length == DT_INFO_INVALID_REG_SIZE) {
+			IMSG("Malformed nvmem %s: ignored", string);
```
[the remark this answers is not in the document]
Comment (GseoC) [reply core/drivers/stm32_bsec.c:705]: Yes, sure, for consistency

File: core/include/drivers/stm32_bsec.h (hunk 1 above)
Comment (GseoC) [reply core/include/drivers/stm32_bsec.h:154]: Better :)

**Comment by GseoC** [comment]
Force-pushed with comments addressed and tags applied

**Pull request merged**